### DISCUSSION TITLE
Better step edit

### DISCUSSION THREAD

**Discussion by maks** [bangcorrupt/hacktribe#83] [Ideas]
Not  sure how possible it is to make modifications to step edit mode, but ... I've recently been trying to learn about tracker workflow using a headless M8 and realised that if you kinda squint, the electribes much maligned step edit mode is somewhat a tracker style sequencer, just with very limited editing functions and only 1 instrument per "column", which I actually prefer to trying to squeeze multiple in instruments together on the same column, while on plus side being able to have 4 note paraphonic on each.

So thinking of it that way, some really nice improvements for step edit mode would be being able to easily select several steps for edit or copy-paste by holding down the pads for those  steps. Also using the last selected note for each new step added would be nicer too. 

There's probably other small improvements like these that could actually I think turn step edit into more powerful tracker-like sequencer, that's assuming it's even possible to hack changes to step edit in the factory firmware.

**Comment by bangcorrupt** [comment]
> easily select several steps for edit or copy-paste by holding down the pads for those steps

This would be really nice.

Anything involving changes to the UI or remapping controls is definitely a long way off, I tried and failed a few times.  I'd like to add all of the motion sequences and maybe groove templates to the step editor.  If I can get that working it could open the door to an onboard FX preset editor.  I think adding items to existing menus will probably be easier than remapping controls, but I've made no progress on either.

**Reply by ptowns** [reply]
Anything with the tie and the gate? Dosnt ever seem to work 100% right for me. In a dream world a tied note would also guide! 😉

**Reply by bangcorrupt** [reply]
> tie and the gate? Dosnt ever seem to work 100% right for me

Can you explain how, or share a pattern demonstrating the issue?  Gate times >96 are tied notes.  If you use the XY pad to input notes, it will never tie.  You would need to add ties in Step Edit.  

> guide

Glide?  Shift+Pitch will set glide time, but it applies to all notes.  It can be motion sequenced to only effect certain steps, but it doesn't help with live playing.  Proper portamento would be nice.

**Comment by awchx** [comment]
A bunch of additional ideas of how to improve the step edit mode:

1. CUT parameter. Sometimes an option to unconditionally cut the note, along with its decay/release parameter, would be great for genres such as electro house or hip hop. Could be an option per-note or in Gate Time. Actually, it's present in trackers. Could be also accessible outside of the Step Edit mode for playing live, using some unused button combo - would be treated as a note and subject to Part Mute and Erase along with other notes.
2. Utilizing pads for note entry, similar to the Keyboard mode. Could be switchable between it and the default mode.
3. Making use of the buttons above the pads and their LEDs.
• Sequencer and Keyboard. This would be the way of changing modes from the default behavior (Sequencer) to the note entry (Keyboard), indicated by their respective LEDs. In the keyboard mode, the main knob would scroll through steps.
• Part Erase. A shortcut to remove notes from the 4-voice note view. Would be a good alternative to turning the knob to the left until the ---- shows up, potentially prolonging its lifespan.
• Part Mute. Might be a shortcut to the aforementioned CUT parameter.
• Trigger. The quick audition of the current step, would be useful in the note entry (keyboard) mode.
4. Maybe engaging the X-axis of the XY pad to quickly enter notes would also work, for example while holding the Touch Scale button to avoid misclicks.
5. More data visible at once on the display. The real estate is there, so for example merging Note, Gate Time and Velocity into a single screen would be convenient, although I'm aware it might be too much without the access to the source code.
